neo4j/internal/bolt/bolt4.go

```diff
@@ -874,13 +886,25 @@ func (b *bolt4) Reset() {
 	}
 }
 
-func (b *bolt4) GetRoutingTable(context map[string]string, bookmarks []string, database string) (*db.RoutingTable, error) {
+func (b *bolt4) GetRoutingTable(context map[string]string, bookmarks []string, database, impersonatedUser string) (*db.RoutingTable, error) {
 	if err := b.assertState(bolt4_ready); err != nil {
 		return nil, err
 	}
 
 	b.log.Infof(log.Bolt4, b.logId, "Retrieving routing table")
-	if b.minor > 2 {
+	if b.minor >= 4 {
+		b.out.appendRouteWithDbContext(context, bookmarks, routeDatabaseContext(database, impersonatedUser))
+		b.out.send(b.conn)
+		succ := b.receiveSuccess()
+		if b.err != nil {
+			return nil, b.err
+		}
+		routingTable := succ.routingTable
+		b.databaseName = routingTable.Database
+		b.impersonatedUserHomeDb = routingTable.Database
```

Comment:
The `db` field in the response will contain the database for which the RT has been queried. This is not necessarily the home database of the user. However, it will certainly be their home db, when no database has been specified.

Reply:
so something like:
```go
if database != db.DefaultDatabase {
    b.impersonatedUserHomeDb = routingTable.Database
}
```
?
the routing table DB is still the database to specify in subsequent operations, right?